contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3285,15 +3285,46 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 								{
 									RoleSpec   *rolspec = lfirst(item);
 									char	   *user_name;
+									char       *db_principal;
+									const char *db_owner_name;
+									int	      	role_oid;
+									int       	rolename_len;
+									char	   *logical_role_name = NULL;
 
 									user_name = get_physical_user_name(db_name, rolspec->rolename, false);
+									db_owner_name = get_db_owner_name(get_cur_db_name());
+									role_oid = get_role_oid(user_name, true);
+									logical_role_name = rolspec->rolename;
+									rolename_len = strlen(logical_role_name);
+
+									if (drop_user)
+										db_principal = "user";
+									else
+										db_principal = "role";
+
+									/* If user is dbo or role is db_owner, restrict dropping */
+									if ((drop_user && strncmp(logical_role_name, "dbo", rolename_len) == 0) ||
+										(drop_role && strncmp(logical_role_name, "db_owner", rolename_len) == 0))
```

Comment:
What if this check is run against user name `d` or `db_o` or `db_ow` etc.? We need to have the length comparison before using strncmp().

Reply:
Please add such testcases.